vnext/Microsoft.ReactNative.Cxx/CoreApp.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+
+struct RNCoreApp {
+  /// default value: L"index.windows"
+  wchar_t const *jsBundleFile;
+  /// default value: L"ms-appx:///Bundle/"
+  wchar_t const *bundleRootPath;
+  /// Provide the name of your React component here
+  wchar_t const *componentName;
+
+  /// default value: true
+  bool useWebDebugger;
+  /// default value: true
+  bool useFastRefresh;
+  /// default value: true
+  bool useDeveloperSupport;
+  /// default value: false
+  bool useDirectDebugger;
+  /// default value: true
+  bool requestInlineSourceMap;
+
+  /// default value: false
+  bool enableDefaultCrashHandler;
+  /// default value: 9229
+  unsigned short debuggerPort;
+  /// default value: 8081
+  unsigned short sourceBundlePort;
+  /// default value: L"localhost"
+  wchar_t const *sourceBundleHost;
+
+  /// default value: "chakra", possible values: "chakra", "hermes"
+  wchar_t const *jsEngine;
+
+  /// default value: nullptr
+  wchar_t const *viewName;
+
+  void *resourcesAbi;
+  /// Provides the launch arguments back to the CoreApp callback
+  wchar_t const *args;
+  void **packageProvidersAbi;
+  unsigned char packageProvidersAbiCount;
+  void *propertiesAbi;
+};
+
+#ifdef __cplusplus
+#define NORETURN [[noreturn]]
+#else
+#define NORETURN _Noreturn
```

Comment:
>_Noreturn [](http://example.com/codeflow?start=17&length=9)

The doc says that it is deprecated in C23 in favor of [[noreturn]].
I guess the condition above must say that [[noreturn]] is used with C++ or with C starting from C23.
[https://en.cppreference.com/w/c/language/_Noreturn](https://en.cppreference.com/w/c/language/_Noreturn)

Reply:
MSVC does not support any C standard beyond C98.